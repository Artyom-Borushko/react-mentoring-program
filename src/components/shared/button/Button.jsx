import './button.css';
import React from "react";

export default function Button({type, children, onClick, disabled, classNames}) {

    return (
        <button
            type={type}
            className={`button ${type} ${disabled ? 'disabled' : ''} ${classNames || ''}`}
            onClick={onClick}
            disabled={disabled}
        >
            {children}
        </button>
    )
}
