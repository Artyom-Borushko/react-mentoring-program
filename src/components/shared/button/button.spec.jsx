import { render, screen } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom';
import Button from './Button';

describe('Button component tests', () => {
  it('renders Button component with children passed to it', () => {
    const { container } = render(<Button>test text</Button>);

    expect(container).toMatchSnapshot();
    expect(screen.getByText('test text')).toBeInTheDocument();
  });
});
