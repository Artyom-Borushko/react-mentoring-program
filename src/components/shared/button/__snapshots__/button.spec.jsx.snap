// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component tests renders Button component with children passed to it 1`] = `
<div>
  <button
    class="button"
  >
    test text
  </button>
</div>
`;
